torchao/prototype/parq/quant/uniform_torchao.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional, Union
+
+import torch
+from torch import Tensor
+
+from torchao.core.config import AOBaseConfig
+from torchao.quantization.quant_primitives import (
+    _DTYPE_TO_BIT_WIDTH,
+    _DTYPE_TO_QVALUE_BOUNDS,
+    MappingType,
+    ZeroPointDomain,
+    choose_qparams_affine,
+    dequantize_affine,
+    quantize_affine,
+)
+
+from .quantizer import Quantizer
+
+_BIT_WIDTH_TO_DTYPE = {v: k for k, v in _DTYPE_TO_BIT_WIDTH.items()}
+
+
+class UnifTorchaoQuantizer(Quantizer):
+    """Uniform quantizer that uses torchao's quantization primitives"""
+
+    def __init__(
+        self,
+        symmetric: bool,
+        target_dtype: Optional[torch.dtype] = None,
+        quant_min: Optional[Union[int, float]] = None,
+        quant_max: Optional[Union[int, float]] = None,
+        eps: Optional[float] = None,
+        preserve_zero: bool = True,
+        zero_point_domain: ZeroPointDomain = ZeroPointDomain.NONE,
+        config: Optional[AOBaseConfig] = None,
```

Comment:
Hmm, a bit weird that this isn't used inside the class at all. Can we just let the user pass this into `torchao_quantize_` instead?